pkg/gds/client/profile_test.go

```diff
@@ -0,0 +1,29 @@
+package client_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/trisacrypto/directory/pkg/gds/client"
+)
+
+// Test that OpenLevelDB opens a LevelDB database.
+func TestOpenLevelDB(t *testing.T) {
+	// No DB specified
+	profile := &client.Profile{}
+	_, err := profile.OpenLevelDB()
+	require.Error(t, err)
+
+	// Invalid DB scheme
+	profile = &client.Profile{DatabaseURL: "foo://bar"}
+	_, err = profile.OpenLevelDB()
+	require.Error(t, err)
+
+	// Valid DB DSN
+	profile = &client.Profile{DatabaseURL: "leveldb:///foo"}
+	db, err := profile.OpenLevelDB()
+	require.NoError(t, err)
+	defer db.Close()
+	defer os.RemoveAll("foo")
```

Comment:
Since this is actually creating a leveldb could we modify this to use a tmpdir (its likely to have more success on container based tests like those run by GitHub actions):

```go
dbpath, err := os.MkdirTemp("", "testdb")
require.NoError(t, err)
defer os.RemoveAll(dbpath)

profile := &client.Profile{DatabaseURL: fmt.Sprintf("leveldb:///%s", dbpath)}
db, err := profile.OpenLevelDB()
require.NoError(t, err)
defer db.Close()
```

Reply:
Note that `fmt.Sprintf("leveldb:///%s", dbpath)` will work if `dbpath` is an absolute path, which I'm guessing it is.